Tests/scripts/add_secrets_file_to_build.py

```diff
@@ -1,27 +1,160 @@
+    labels_filter_master = {'pack_id': '!=None', 'ignore': '==None', 'merged': '==None', 'dev': '!=None',
+                            'branch_name': f'=="{branch_name}"'}
+    master_secrets = secret_conf.list_secrets(options.gsm_project_id, labels_filter_branch, name_filter=yml_pack_ids,
+                                              with_secrets=True)
+    branch_secrets = secret_conf.list_secrets(options.gsm_project_id, labels_filter_master, with_secrets=True)
+
+    if branch_secrets:
+        for dev_secret in branch_secrets:
+            replaced = False
+            for i in range(len(master_secrets)):
+                if dev_secret['name'] == master_secrets[i]['name']:
+                    master_secrets[i] = dev_secret
+                    replaced = True
+                    break
+            # If the dev secret is not in the changed packs it's a new secret
+            if not replaced:
+                master_secrets.append(dev_secret)
+
     secret_file = {
         "username": options.user,
         "userPassword": options.password,
-        "integrations": secrets
+        "integrations": master_secrets
     }
+    return secret_file
 
+
+def write_secrets_to_file(options: argparse.Namespace, secrets: dict):
+    """
+    Writes the secrets we got from GSM to a file for the build
+    :param options: the parsed parameter for the script
+    :param secrets: a list of secrets to be used in the build
+    """
     with open(options.json_path_file, 'w') as secrets_out_file:
         try:
-            secrets_out_file.write(json5.dumps(secret_file, quote_keys=True))
+            secrets_out_file.write(json5.dumps(secrets, quote_keys=True))
         except Exception as e:
             logging.error(f'Could not save secrets file, malformed json5 format, the error is: {e}')
     logging.info(f'saved the json file to: {options.json_path_file}')
 
 
-def options_handler(args=None):
+def get_yml_pack_ids(changed_packs: list[str]) -> list[str]:
+    """
+    Gets the changed integration IDs from the YML file
+    :param changed_packs: a list of changed packs in the current branch
+    :return: the list of IDs of integrations to search secrets for
+    """
+    yml_ids = []
+    for changed_pack in changed_packs:
+        root_dir = Path(changed_pack)
+        root_dir_instance = pathlib.Path(root_dir)
+        yml_files = [item.name for item in root_dir_instance.glob("*") if str(item.name).endswith('yml')]
+        for yml_file in yml_files:
+            with open(f'{changed_pack}/{yml_file}', "r") as stream:
+                try:
+                    yml_obj = yaml.safe_load(stream)
+                    yml_ids.append(yml_obj['commonfields']['id'])
+                except yaml.YAMLError as exc:
+                    logging.error(f'Could not extract ID from {yml_file}: {exc}')
+    return yml_ids
+
+
+def get_changed_packs(changed_files: list[str]) -> list[str]:
```

Comment:
This seems like it could better use a generic function such as `find_pack_folder`

Reply:
I've added it